DQM/HLTEvF/plugins/TriggerRatesMonitor.cc

```diff
@@ -191,19 +188,35 @@ void TriggerRatesMonitor::dqmBeginRun(edm::Run const &run,
       histograms.hlt_by_dataset_counts[i].resize(paths.size());
       histograms.datasets[i].reserve(paths.size());
       for (auto const &path : paths) {
-        histograms.datasets[i].push_back(histograms.hltConfig.triggerIndex(path));
+        auto const triggerIdx = histograms.hltConfig.triggerIndex(path);
+        if (triggerIdx < nTriggers)
+          histograms.datasets[i].push_back(triggerIdx);
```

Comment:
This (i.e. the check on `< nTriggers`) is the change to fix the error reported in [CMSHLT-2577](https://its.cern.ch/jira/browse/CMSHLT-2577). A similar check can be found in
https://github.com/cms-sw/cmssw/blob/7e6539b24427faa92b1411a9903202169a3c3990/HLTrigger/JSONMonitoring/plugins/HLTriggerJSONMonitoring.cc#L230-L232

The rest of the changes in this PR is about adding a few checks, plus minor cleanup.